src/Pages/DetailProduct/DetailProduct.js

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import { withRouter } from "react-router-dom";
+import "./DetailProduct.scss";
+
+class DetailProduct extends React.Component {
+  state = {
+    input: 1,
+    price: 6600,
+    totalPrice: 6600,
+  };
+
+  subtractHandler = (e) => {
+    const { input } = this.state;
+    if (e.target.name === "plus") {
+      this.setState({
+        input: input + 1,
+      });
+    } else if (e.target.name === "minus" && input > 1) {
+      this.setState({
+        input: input - 1,
+      });
+    } else {
+      this.setState({
+        input: 1,
+      });
+    }
+  };
```

Comment:
```suggestion
  subtractHandler = (e) => {
    const { input } = this.state;
    const { name } = e.target;
    const isPlus = name === "plus"
    
    if (!isPlus && input === 1) return 1;
    
    this.setState({
      input: isPlus ? input += 1 : input -= 1
    });
  };
```
위와 같이 고칠 수 있습니다. 함수명도 보다 적절한 이름으로 바꿔주세요.

Reply:
와우 한수 두수 세수 배우고 가겠습니다...